soroban-env-host/src/host.rs

```diff
@@ -2416,8 +2425,10 @@ impl VmCallerEnv for Host {
         let vnew = self.visit_obj(b, |hv: &ScBytes| {
             self.validate_index_lt_bound(i, hv.len())?;
             let mut vnew: Vec<u8> = hv.metered_clone(self)?.into();
-            // len > i has been verified above but use saturating_sub just in case
-            let n_elts = (hv.len() as u64).saturating_sub(i as u64);
+            // len > i has been verified above but use checked_sub just in case
+            let n_elts = (hv.len() as u64)
+                .checked_sub(i as u64)
+                .ok_or_else(|| self.err_arith_overflow())?;
```

Comment:
This should be an internal error if we claim to guarantee the invariant in the comment above.

Reply:
Good point, fixed.